### DISCUSSION TITLE
File upload field

### DISCUSSION THREAD

**Discussion by ChristopherNowlan** [payloadcms/plugin-form-builder#38] [Ideas]
This isn't really an issue but a feature request/query. Is there a native file upload field yet or in the pipeline?

**Comment by jacobsfletch** [comment]
Not natively but you can easily add in your own custom upload field to the form using the `fields` property on your config. If you have success with this, we can definitely add it into the default config! I think that's a great idea.

**Comment by jacobsfletch** [comment]
I'm going to go ahead and convert this to a discussion in the "Ideas" category so we can continue the conversation!

**Comment by zanami** [comment]
This would be neat. I've just been through "everything is great but could you also add a file attachment field to that form, just in case" situation.

So, does this require something like

1. Another collection with [upload:true](https://payloadcms.com/docs/upload/overview#enabling-uploads) to hold files, say, SubmittedFiles
2. A custom field/block to configure this relation (name: upload, relationTo:"SubmittedFiles" etc.) 
3. A component on the front end to display and [handle upload](https://payloadcms.com/docs/upload/overview#uploading-files) to SubmittedFiles, populate the data for FormSubmissions...
5. A field in FormSubmissions (like [Upload](https://payloadcms.com/docs/fields/upload), or group like 'payment', or JSON...)
6. sendEmail hook to support attachments

Some thoughts
(3) here is going to take a lot of effort and it's somewhat out of scope of the plugin itself.
Uploads will probably require something like /pages/api/upload on the front end to prevent abuse, as create endpoint should be publicly available

Am I missing something? Easier way?

**Reply by treturner** [reply]
I've implemented my own upload field and I completely agree with the perspective @zanami provided. I have an "upload field" which is just another label + value pair like the others. Biggest thing is handling all of the logic on the frontend as I submit my form data, and then submit the image to a different endpoint on my server.

**Reply by PikaYuhno** [reply]
Do you mind sharing the code for it?
